docs/narr/urldispatch.rst

```diff
@@ -908,12 +908,28 @@ exactly the same job:
        config.add_route('hasslash', 'has_slash/')
        config.scan()
 
-.. warning::
-
-   You **should not** rely on this mechanism to redirect ``POST`` requests.
-   The redirect  of the slash-appending :term:`Not Found View` will turn a
-   ``POST`` request into a ``GET``, losing any ``POST`` data in the original
-   request.
+You **should not** rely on the default mechanism to redirect ``POST`` requests.
+The redirect of the slash-appending :term:`Not Found View` will turn a ``POST``
+request into a ``GET``, losing any ``POST`` data in the original request.  But
+if the argument supplied as ``append_slash`` is the special object
+:attr:`~pyramid.views.UseSubrequest`, a :term:`subrequest` will be issued
+instead of a redirect.  This makes it possible to successfully invoke a
+slash-appended URL without losing the HTTP verb, POST data, or any other
+information contained in the original request.  Instead of returning a redirect
+response when a slash-appended route is detected during the not-found
+processing, Pyramid will call the view associated with the slash-appended route
+"under the hood" and will return whatever response is returned by that view.
+This has the potential downside that both URLs (the slash-appended and the
+non-slash-appended URLs) in an application will be "canonical" to clients; they
+will behave exactly the same, and the client will never be notified that the
+slash-appended URL is "better than" the non-slash-appended URL by virtue of a
+redirect.  It, however, has the upside that a POST request with a body can be
+handled successfully with an append-slash during notfound processing.
```

Comment:
Is it "notfound" or "not-found" processing? Both spellings are used in this paragraph. Please choose one.